apps/website/src/routes/docs/headless/progress/examples/hero.tsx

```diff
@@ -0,0 +1,28 @@
+import { component$, useSignal, useVisibleTask$ } from '@builder.io/qwik';
+import * as Progress from '@qwik-ui/headless';
+
+export default component$(() => {
+  const progress = useSignal(30);
+
+  useVisibleTask$(() => {
```

Comment:
I  thought this restriction was just in the implementation, not in the documentation. But makes sense to avoid it in the docs.

```tsx
  useOnWindow(
    'load',
    $(() => {
      setTimeout(() => {
        progress.value = 50;
      }, 1000);
    }),
  );
```

This is a better approach?

Reply:
> I thought this restriction was just in the implementation, not in the documentation. But makes sense to avoid it in the docs.
> 
> ```tsx
>   useOnWindow(
>     'load',
>     $(() => {
>       setTimeout(() => {
>         progress.value = 50;
>       }, 1000);
>     }),
>   );
> ```
> 
> This is a better approach?

You're correct, for those consuming the library won't be an issue. 

It'll be more of a docs issue.

This would be the preferred approach:
```tsx
  useTask$(() => {
      setTimeout(() => {
        progress.value = 50;
      }, 1000);
  });
```

Since you can execute it on the server. A visible task would make more sense in the case of a viewport animation, because visible task uses the `qvisible` event under the hood. Most things are tied to events in some way! 

A question here would be, shouldn't the user perform some interaction before their progress goes up? Whether that's clicking a button, scrolling, etc.

We want to delay the execution of code for as long as possible, and the code is then executed from the browser cache in a separate thread, kinda like video buffering!

Side note: is there an animation that's supposed to play? I noticed a transition on the headless component, but there is no animation.

Reply:
Yes, it has a small animation, but I removed it from the headless doc and only left it on the styled to make it more direct. 

In the styled I maintained the useOnWindow because if I use useTask it runs on the server, not allowing me to see the transition animation.